Makefile

```diff
@@ -55,11 +52,21 @@ help: ## Display this help.
 	@printf "$$(git rev-parse HEAD 2>/dev/null)" >$@
 	@test -n "$$(git status --porcelain --untracked-files=no)" && echo -dirty >>$@ || true
 
-agent-protocol-forwarder: .git-commit $(SOURCES)
-	$(GOOPTIONS) CGO_ENABLED=0 go build $(GOFLAGS) -ldflags="-X 'github.com/confidential-containers/cloud-api-adaptor/cmd.VERSION=$(shell echo "unknown")' -X 'github.com/confidential-containers/cloud-api-adaptor/cmd.COMMIT=$(shell cat .git-commit)'" -o "$@" "cmd/$@/main.go"
+GOFLAGS += -ldflags="-X 'github.com/confidential-containers/cloud-api-adaptor/cmd.VERSION=$(shell echo "unknown")' \
+                     -X 'github.com/confidential-containers/cloud-api-adaptor/cmd.COMMIT=$(shell cat .git-commit)'"
 
-cloud-api-adaptor: .git-commit $(SOURCES)
-	$(GOOPTIONS) go build $(GOFLAGS) -ldflags="-X 'github.com/confidential-containers/cloud-api-adaptor/cmd.VERSION=$(shell echo "unknown")' -X 'github.com/confidential-containers/cloud-api-adaptor/cmd.COMMIT=$(shell cat .git-commit)'" -o "$@" "cmd/$@/main.go"
+# Build tags required to build cloud-api-adaptor are derived from BUILTIN_CLOUD_PROVIDERS.
+# When libvirt is specified, CGO_ENABLED is set to 1.
+space := $() $()
+comma := ,
+cloud-api-adaptor: GOFLAGS += -tags $(subst $(space),$(comma),$(BUILTIN_CLOUD_PROVIDERS))
+
+ifneq (,$(filter libvirt,$(BUILTIN_CLOUD_PROVIDERS)))
+cloud-api-adaptor: GOOPTIONS := $(subst CGO_ENABLED=0,CGO_ENABLED=1,$(GOOPTIONS))
+endif
+
```

Comment:
The `filter` function accepts whitespace-separated words as an input.
https://www.gnu.org/software/make/manual/html_node/Text-Functions.html

Reply:
`findstring` would fit better here no?

Reply:
`filter` recognizes whitespace-separated words, so it does exact matches. `findstring` just searches a substring, so incorrect matches may occur.